### DISCUSSION TITLE
Working together with SQLGlot?

### DISCUSSION THREAD

**Discussion by tobymao** [macbre/sql-metadata#329] [General]
I just stumbled upon this cool library and it seems there may be a lot of overlap with my project SQLGlot. Correct me if i’m wrong but my guess is this project was made because sqlparse only tokenizes queries and doesn’t create an ast so that you can do things like extract column names easily.

i have an example of how to do that here

https://github.com/tobymao/sqlglot#metadata

Anyways, just wanted to start a friendly discussion and see if we can possibly work together.

**Comment by collerek** [comment]
Hi! Saw your project some time ago and was about to check it out but forgot. Yesterday I watched your talk at data+ai summit and starred your repo so I remember to check it out ;) I don't know if this is how you end up here or is it just a coincidence 😁 

Yeah, you are basically right, sqlparse has its limitations. We managed to overcome some of them, but keeping the query elements in flat data structures reached a point when it was bringing more complications than we were ready to accept. We were thinking about rewriting the parser in a more structured way but we didn't have time for this (hence not many changes in our repo in recent weeks).

I guess we are ready to cooperate, that's the whole idea of open-source software anyway, right?! 😄 
How would you see us working together?

**Comment by macbre** [comment]
Hi, Toby!

I'm all in for the cooperation between us! Repeating @collerek question here - how would you see us working together?

**Comment by tobymao** [comment]
@collerek, yea, I found this repo when I saw you starred it. 

As a first step, I think it'd greatly simplify this project if you used SQLGlot to parse things. I guess I can start it off by making a PR to show how this could be done?

**Comment by macbre** [comment]
> I guess I can start it off by making a PR to show how this could be done?

Definitely!

**Comment by tobymao** [comment]
i see this is in now :)
